service/telemetry.go

```diff
@@ -78,10 +78,13 @@ type telemetryInitializer struct {
 }
 
 func newColTelemetry(registry *featuregate.Registry) *telemetryInitializer {
-	registry.MustRegister(featuregate.Gate{
-		ID:          useOtelForInternalMetricsfeatureGateID,
-		Description: "controls whether the collector to uses OpenTelemetry for internal metrics",
-		Enabled:     false,
+	// Ensure we only do the register once
+	internalMetricFGRegisterOnce.Do(func() {
```

Comment:
It feels awkward to have a once-ever command occur within what is essentially a constructor. Should this just be moved elsewhere? Perhaps in an `init` func?

This might allow the test override to be removed as well.

Reply:
Yeah, it's not uncommon for singleton patterns to do this in getters that act as first time creations. Though the purpose of this function has changed with the new logic. I have no problem putting it into an init function. I will have to double check the tests though. If this logic needs to be tested it'll have to be broken out into it's own function that is wrapped in an init.

Reply:
Pushed a refactor to remove the `Once` and move the register into an init function.